hive template sample

spelluru opened the pull request:
Azure Data Factory - Hive Transformation template

azurecla commented:
Hi __@spelluru__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (spelluru). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

spelluru commented:
@bmoore-msft - Hi Brian, here is my other PR with a template that needs to be published. Please review when you get a chance. Thanks.

bmoore-msft commented on 101-data-factory-hive-transformation/README.md:
```diff
@@ -0,0 +1,55 @@
+
+For more information, see [Overview and prerequisites](https://azure.microsoft.com/documentation/articles/data-factory-build-your-first-pipeline/) article. 
+
+## Deploy using PowerShell
```
update this section to use the script provided in the root of the repo

spelluru replied:
Fixed.

bmoore-msft commented on 101-data-factory-hive-transformation/azuredeploy.json:
```diff
@@ -0,0 +1,207 @@
+                    "scriptPath": "[concat(parameters('blobContainer'), '/', parameters('hiveScriptFolder'), '/', parameters('hiveScriptFile'))]",
+                    "scriptLinkedService": "[variables('azureStorageLinkedServiceName')]",
+                    "defines": {
+                      "inputtable": "[concat('wasb://', parameters('blobContainer'), '@', parameters('storageAccountName'), '.blob.core.windows.net/', parameters('inputBlobFolder'))]",
```
see #9 in BP guide

spelluru replied:
"inputtable" should be evaluated to: 
"inputtable": "https://changemeazurestorage.blob.core.windows.net/adfgetstarted/inputdata",
"partitionedtable": "https://changemeazurestorage.blob.core.windows.net/adfgetstarted/partitioneddata"

where changemeazurestorage is the name of the storage. Users change this name in the parameter file. As a prerequisite to deploying the template, user has to create inputdata folder in the adfgetstarted container and upload the input file (input.log) to the inputdata folder. 
The partitioneddata folder is automatically created when the Hive script runs. 

As per the best practice, I changed the above code to:
"inputtable": "[concat(reference(concat('Microsoft.Storage/storageAccounts/', parameters('storageAccountName')), '2015-06-15').primaryEndpoints.blob,  parameters('blobContainer'), '/',parameters('inputBlobFolder'))]",
"partitionedtable": "[concat(reference(concat('Microsoft.Storage/storageAccounts/', parameters('storageAccountName')), '2015-06-15').primaryEndpoints.blob,  parameters('blobContainer'), '/',parameters('outputBlobFolder'))]"

I can deploy the template locally from machine from Azure CLI successfully since the changemeazurestorage exists in the resource group I specify as a parameter. 

However, when I check in the code to GIT, the validation fails as the changemeazurestorage does not exist in the resource group that build uses. 

I appreciate if you can help resolve this issue.

spelluru replied:
@bmoore-msft - can you please help with resolving the issue? Thanks!

spelluru replied:
@bmoore-msft - 

Storage account 
        adfgetstarted (container)
                inputdata (folder) contains the input web log file. 
                script (folder) contains the Hive script to process the input log file. 
                partitioneddata (folder) contains the output partitioned data

I could create a container with inputdata and script folders and give them the public read access. Then the syntax in the file should work. But, writing to a partitioneddata is not possible with a public blob.

bmoore-msft commented on 101-data-factory-hive-transformation/azuredeploy.json:
```diff
@@ -0,0 +1,207 @@
+                    "scriptLinkedService": "[variables('azureStorageLinkedServiceName')]",
+                    "defines": {
+                      "inputtable": "[concat('wasb://', parameters('blobContainer'), '@', parameters('storageAccountName'), '.blob.core.windows.net/', parameters('inputBlobFolder'))]",
+                      "partitionedtable": "[concat('wasb://', parameters('blobContainer'), '@', parameters('storageAccountName'), '.blob.core.windows.net/', parameters('outputBlobFolder'))]"
```
see #9 in BP guide

spelluru replied:
Please see the comment above.

bmoore-msft commented on 101-data-factory-hive-transformation/azuredeploy.json:
```diff
@@ -0,0 +1,207 @@
+                {
+                  "type": "HDInsightHive",
+                  "typeProperties": {
+                    "scriptPath": "[concat(parameters('blobContainer'), '/', parameters('hiveScriptFolder'), '/', parameters('hiveScriptFile'))]",
```
Can we make this part of the sample?  It's hard to tell what it's doing otherwise...

spelluru replied:
The scriptPath gets resolved to adfgetstarted/script/partitionweblogs.hql. A prerequisite for this sample is to create **adfgetstarted** container, **script** folder, and upload the **partitionweblogs.hql** file to this folder.

spelluru commented:
I still get the error though. 

I added the following parameter: 
    "storageAccountResourceGroupName": {
"value": "ADFTutorialResourceGroup"
    },   

Updated the inputdata and partitionedata expressions: 

"inputtable": "[concat(replace(reference(resourceId(parameters('storageAccountResourceGroupName'), 'Microsoft.Storage/storageAccounts/', parameters('storageAccountName')), '2016-01-01').primaryEndpoints.blob, 'https://', concat('wasb://', parameters('blobContainer'), '@')), parameters('inputBlobFolder'))]",
"partitionedtable": "[concat(replace(reference(resourceId(parameters('storageAccountResourceGroupName'), 'Microsoft.Storage/storageAccounts/', parameters('storageAccountName')), '2016-01-01').primaryEndpoints.blob, 'https://', concat('wasb://', parameters('blobContainer'), '@')), parameters('outputBlobFolder'))]"                       

I can deploy it fine on my machine using Azure CLI. However, when I check-in the code, I get the following error: 
error: Deployment provisioning state was not successful\nerror: Resource group 'ADFTutorialResourceGroup' could not be found.

https://travis-ci.org/Azure/azure-quickstart-templates/builds/167437872

Unless this resource group is created in the subscription that the travis build uses, this won’t go through. Please let me know how I can fix it.

spelluru commented:
@bmoore-msft - please see the note above. Thanks.

bmoore-msft commented:
CI failure is coming from the "existing" storage requirement and confirmed that manual deployment works.
